release.py

```diff
@@ -0,0 +1,694 @@
+        VERSION_FILES['ops/pyproject'],
+        testing_version,
+        deps={'ops-scenario': testing_version, 'ops-tracing': ops_version},
+    )
+
+
+def update_testing_version(ops_version: str, testing_version: str):
+    """Update the testing pyproject version."""
+    update_pyproject_versions(VERSION_FILES['testing'], testing_version, deps={'ops': ops_version})
+
+
+def update_tracing_version(ops_version: str):
+    """Update the tracing pyproject version."""
+    update_pyproject_versions(VERSION_FILES['tracing'], ops_version, deps={'ops': ops_version})
+
+
+def update_uv_lock():
+    """Update the uv.lock file with the new versions."""
+    subprocess.run(['uv', 'lock'], check=True)  # noqa: S607
+
+
+def parse_scenario_version():
+    """Parse the current scenario version from pyproject.toml."""
+    file_path = VERSION_FILES['testing']
+    content = file_path.read_text()
+    match = re.search(rf'version = "{VERSION_REGEX}"', content)
+    if not match:
+        raise ValueError(f'Could not find version string in {file_path}')
+    version_str = match.group(1)
+    base_version, dev_suffix = parse_version(version_str)
+    return base_version, dev_suffix
+
+
+def get_new_scenario_version_for_release() -> str:
+    """Get a new version for scenario.
+
+    Default value is generated by removing the ".dev0" suffix from the current scenario version.
+
+    Can be overridden with a user-provided version.
+    """
+    suggested_version, _ = parse_scenario_version()
+    logger.info(f'Suggested new scenario version: {suggested_version}')
+
+    while True:
+        user_input = input(
+            f'Input new scenario version (press enter to use the suggested version '
+            f"{suggested_version or 'required'}), or 'c' to cancel: "
+        ).strip()
+
+        if user_input.lower() == 'c':
+            logger.warning('Scenario version creation canceled.')
+            exit(0)
+
+        suggested_version = user_input if user_input else suggested_version
+
+        if not suggested_version:
+            logger.error('Error: No version specified and no suggestion available')
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+$', suggested_version):
+            logger.error('Error: Version must be in format X.Y.Z')
+            continue
+
+        confirm = (
+            input(f"Confirm using scenario version '{suggested_version}'? [y/N]: ").strip().lower()
+        )
+
+        if confirm == 'y':
+            break
+
+        logger.info("Let's try again...")
+
+    return suggested_version
+
+
+def update_versions_for_release(tag: str):
+    """Update version files to the specified release version."""
+    scenario_version = get_new_scenario_version_for_release()
+    update_ops_version(tag, scenario_version)
+    update_testing_version(tag, scenario_version)
+    update_tracing_version(tag)
+    update_uv_lock()
+
+
+def get_new_version_post_release(repo: github.Repository.Repository, branch_name: str) -> str:
```

Comment:
Can we completely automate this? Won't this always be minor+1 with the `.dev` suffix?

Reply:
This exists specifically for major releases (I made the script during 2.23 -> 3.0 release). If the previous version is 2.23, and now we want to release 3.0, a manual override is required. But I don't think we will make any major release soon, so I will fully automate this.